recipes/default.rb

```diff
@@ -15,3 +15,174 @@
+config_options['$DatabaseType'] = node['osl-rt']['db']['type']
+config_options['$DatabaseHost'] = node['osl-rt']['db']['host']
+config_options['$DatabaseRTHost'] = node['osl-rt']['db']['host']
+config_options['$DatabaseName'] = node['osl-rt']['db']['name']
+config_options['$DatabaseUser'] = node['osl-rt']['db']['username']
+config_options['$DatabasePassword'] = node['osl-rt']['db']['password']
+config_options['_Plugins'] = node['osl-rt']['plugins']
+config_options['_Lifecycles'] = node['osl-rt']['lifecycles']
+
+# Set up the queue emails
+rt_emails = init_emails(node['osl-rt']['queues'], node['osl-rt']['fqdn'], node['osl-rt']['default'])
+
+config_options['$RTAddressRegexp'] = "^(#{rt_emails.join('|')}(-comment)?\@(#{node['osl-rt']['fqdn']}))"
+
+# Set up the hostname as the website
+hostname node['hostname'] do
+  aliases ["#{node['osl-rt']['fqdn']}"]
+end
+
+# Parse the config hash to a string we can use as
+# the RT_SiteConfig.pm file.
+strConfigFile = parse_config(config_options)
+
+# Root Account
+template '/root/.rtrc' do
+  source 'rt/rtrc.erb'
+  cookbook 'osl-rt'
+  user 'root'
+  group 'root'
+  mode '0600'
+  sensitive true
+  variables(root_pass: node['osl-rt']['root-password'], domain: node['osl-rt']['fqdn'])
+end
+
+# Add the RT command to the root user's PATH
+link '/usr/local/sbin/rt' do
+  to '/opt/rt/bin/rt'
+end
+
+# RT Initial Configuration.
+file '/opt/rt/etc/RT_SiteConfig.pm' do
+  content strConfigFile
+  user 'root'
+  group 'apache'
+  mode '0640'
+  sensitive true
+  notifies :reload, 'apache2_service[osuosl]'
+end
+
+# Initalize the DB
+execute 'init-db-rt' do
+  command <<-EOH
+    /opt/rt/sbin/rt-setup-database \
+      --action init \
+      --dba #{config_options['$DatabaseUser']} \
+      --dba-password #{config_options['$DatabasePassword']} \
+      --skip-create && \
+    touch /opt/rt/chef/init-db-rt
+  EOH
+  creates '/opt/rt/chef/init-db-rt'
+  sensitive true
+end
+
+# Set a new password for root
+execute 'Set root password' do
+  command <<-EOH
+    mysql -u #{config_options['$DatabaseUser']} \
+      -p#{config_options['$DatabasePassword']} \
+      -e 'UPDATE Users \
+        SET Password=md5(\"#{node['osl-rt']['root-password']}\") \
+        WHERE Name=\"root\";' \
+      #{config_options['$DatabaseName']} && \
+    touch /opt/rt/chef/init-root-passwd
+  EOH
+  creates '/opt/rt/chef/init-root-passwd'
+  sensitive true
+end
+
+# Set up web app
+apache_app node['osl-rt']['fqdn'] do
+  directory '/opt/rt/share/html'
+  include_config true
+  include_template true
+  include_directory 'rt'
+  include_name 'rt'
+  include_params('domain': node['osl-rt']['fqdn'])
+end
+
+# Forcefully reload Apache during the initial run, in order to allow for setting up the queues properly.
+service 'httpd' do
+  action :reload
+  not_if { ::File.exist?('/var/spool/mail/nobody') }
+end
+
+# Set up the queues in RT
+node['osl-rt']['queues'].each do |pt, email|
+  execute "Creating RT queue for #{pt}" do
+    command <<~EOL
```

Comment:
I like doing "end of command" or `EOC` for execute resources.
```suggestion
    command <<~EOC
```